src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -137,11 +137,12 @@ protected void messageReceivedDecorate(final T request, final TransportRequestHa
 
             //bypass non-netty requests
             if(channelType.equals("direct")) {
-                final String userHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_USER_HEADER);
+                // for direct channel requests we don't serialize the user object in sendRequestDecorate
+                final User user = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
                 final String injectedRolesHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES_HEADER);
                 final String injectedUserHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER_HEADER);
 
-                if(Strings.isNullOrEmpty(userHeader)) {
+                if(user != null) {
```

Comment:
Why remove the empty check here?

Reply:
Because for direct requests we don't pass headers (which would then be serialized and put as a transient), rather we directly put the transient object, User, in the thread context via `sendMessageDecorate` in SecurityInterceptor and retrieve it as an object here. In doing so we skip serialization and deserialization of direct requests.